qiskit_ibm_runtime/runtime_job.py

```diff
@@ -320,12 +323,34 @@ def _set_error_message(self, job_response: Dict) -> None:
         """
         if self._status == JobStatus.ERROR:
             job_result_raw = self._api_client.job_results(job_id=self.job_id)
-            self._error_message = API_TO_JOB_ERROR_MESSAGE[
-                job_response["status"].upper()
-            ].format(self.job_id, job_result_raw)
+            if "reason" in job_response["state"]:
+                error_msg = (
+                    job_response["state"]["status"]
+                    + " - "
+                    + job_response["state"]["reason"]
+                )
+            else:
+                error_msg = job_response["state"]["status"]
+            self._error_message = API_TO_JOB_ERROR_MESSAGE[error_msg.upper()].format(
+                self.job_id, job_result_raw
+            )
```

Comment:
I'm personally not a fan of using string concatenation here as well. It appears to be done only to retrofit the "new" response format on our "old" datastructure.

Why not use the same approach to compute the error messages that you now use to compute the status?

e.g. 

```
self._error_message = _error_message_from_job_response(...)
```

Maybe a simple mapping table is no longer the best approach...

Reply:
I agree, it's best to not use the `API_TO_JOB_ERROR_MESSAGE` constant and just handle the error msg in a separate method